contracts/EverlongStrategy.sol

```diff
@@ -520,6 +546,183 @@ contract EverlongStrategy is BaseStrategy {
         return output;
     }
 
+    // ╭───────────────────────────────────────────────────────────────────────╮
+    // │                         Wrapped Token Helpers                         │
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    /// @dev Wrap the hyperdrive token so that it can be used in the strategy.
+    /// @param _unwrappedAmount Amount of unwrapped hyperdrive tokens to wrap.
+    /// @return _wrappedAmount Amount of wrapped tokens received.
+    function _wrap(
+        uint256 _unwrappedAmount
+    ) internal returns (uint256 _wrappedAmount) {
+        // If the strategy doesn't use a wrapped asset, revert.
+        if (!isWrapped) {
+            revert IEverlongStrategy.AssetNotWrapped();
+        }
+
+        // Approve the wrapped asset contract for the hyperdrive tokens.
+        // Add one to the approval amount to leave the slot dirty and save
+        // gas on future approvals.
+        ERC20(hyperdriveToken).approve(address(asset), _unwrappedAmount);
+
+        // Wrap the hyperdrive tokens.
+        _wrappedAmount = IERC20Wrappable(address(asset)).wrap(_unwrappedAmount);
+    }
+
+    /// @dev Unwrap the strategy asset so that it can be used with hyperdrive.
+    /// @param _wrappedAmount Amount of wrapped strategy assets to unwrap.
+    /// @return _unwrappedAmount Amount of unwrapped tokens received.
+    function _unwrap(
+        uint256 _wrappedAmount
+    ) internal returns (uint256 _unwrappedAmount) {
+        // If the strategy doesn't use a wrapped asset, revert.
+        if (!isWrapped) {
+            revert IEverlongStrategy.AssetNotWrapped();
+        }
+
+        // Unwrap the strategy assets.
+        _unwrappedAmount = IERC20Wrappable(address(asset)).unwrap(
+            _wrappedAmount
+        );
+    }
+
+    /// @dev Calculate the maximum amount of strategy assets that can be spent
+    ///      on longs before hyperdrive runs out of liquidity.
+    /// @return assets Max amount of strategy assets that can be spent on longs.
+    function _calculateMaxLong() internal view returns (uint256 assets) {
+        // Retrieve the max long amount denominated in hyperdrive base tokens.
+        assets = IHyperdrive(hyperdrive).calculateMaxLong();
+
+        // If `asBase` is false, convert the amount to be denominated in vault
+        // share tokens.
+        if (!asBase) {
+            assets = IHyperdrive(hyperdrive)._convertToShares(assets);
+        }
+    }
+
+    /// @dev Open a long with the specified amount of assets. Return the amount
+    ///      of bonds received and their maturityTime.
+    /// @param _toSpend Amount of strategy assets to spend.
+    /// @param _minOutput Minimum amount of bonds to accept.
+    /// @param _minVaultSharePrice Minimum hyperdrive vault share price to
+    ///                            purchase at.
```

Comment:
Nit: 

```suggestion
    /// @param _minVaultSharePrice Minimum hyperdrive vault share price to
    ///                purchase at.
```

Reply:
I know it looks weird but this is actually consistent with how I've done commenting for parameters in the everlong repo (which I'm just now realizing is different than hyperdrive).

I like how hyperdrive does it but feel as though this should maybe be left as is and all parameter comments updated at once in a future pr.

Reply:
Sounds good. Not a big deal, and I'm a fan of keeping things consistent.